components/empty-state/src/empty-state.ts

```diff
@@ -235,3 +235,12 @@ export class DtEmptyState
     }
   }
 }
+
+/**
+ * Marks a custom empty state component to be used within, for example, `<dt-table>`.
+ * The custom empty state must both extend and provide itself as a DtEmptyState.
+ */
+@Directive({
+  selector: '[dtEmptyState]',
+})
+export class DtEmptyStateDirective {}
```

Comment:
In the library we are not using the `Directive` suffix. 
See https://barista.dynatrace.com/components/coding-standards/#classes

Also pls rename the **class** and the **attribute selector** to something like `DtEmptyStateCustom`/`[dtEmptyStateCustom]` or `DtCustomEmptyState`/`[dtCustomEmptyState]` so it is clear that this is a different way of providing an empty-state and the consumer does not get confused with emty-state vs custom-emty-state.